optuna/samplers/_tpe/sampler.py

```diff
@@ -412,7 +412,7 @@ def _split_observation_pairs(
         self, config_vals: Dict[str, List[Optional[float]]], loss_vals: List[Tuple[float, float]]
     ) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray]]:
 
-        config_values = {k: np.asarray(v, dtype=float) for k, v in config_vals.items()}
+        config_values = {k: np.asarray(v) for k, v in config_vals.items()}
```

Comment:
I guess we can add a unit test to confirm that this change resolve the error caused by the `NaN` cast. What do you think of it?

Reply:
Sounds reasonable. Maybe even in `tests/samplers_tests/test_samplers.py`? 

It's by the way trivial to reproduce. E.g.

```python
import optuna as ot


def obj(t):
    x = t.suggest_float("x", 0, 1)
    if t.number % 2 == 0:
        _ = t.suggest_float("y", 0, 1)
    return x


if __name__ == "__main__":
    sampler = ot.samplers.TPESampler(n_startup_trials=1)
    study = ot.create_study(sampler=sampler)
    study.optimize(obj, n_trials=10)
```